### ISSUE TITLE
Update/overhaul accessibility testing

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#7366]
Definition of done per component: 
- convert existing tests to React Testing Library
- add Accessibility Checker/Axe tests
- determine and implement missing WCAG 2.1 tests
- determine and document manual testing practices
---

- [x] Accordion
- [ ] Breadcrumb
- [ ] Button
- [ ] Checkbox
- [ ] Code snippet
- [ ] Content switcher
- [ ] Data table 
- [ ] Date picker
- [ ] Dropdown 
- [ ] File uploader
- [ ] Form
- [ ] Inline loading
- [x] Link
- [ ] List
- [ ] Loading
- [ ] Modal
- [ ] Notification
- [ ] Number input
- [ ] Overflow menu
- [ ] Pagination
- [ ] Progress indicator
- [ ] Radio button
- [ ] Search
- [ ] Select
- [ ] Slider
- [ ] Structured list
- [ ] Tabs
- [ ] Tag
- [ ] Text input
- [ ] Tile
- [ ] Toggle
- [ ] Tooltip
- [ ] UI Shell
